project_structure.txt

```diff
@@ -0,0 +1,119 @@
+.modin/
+   .config/
+   .distributed/
+   .pandas/
+   .test/
+   .rest files
+   .core/
+      .io/
+         .column_stores/
+            . # the same as it is now
+         .sql/
+            . # the same as it is now
+         .text/
+            . # the same as it is now
+         .file_dispatcher.py # the same as it is now
+         .io.py # the same as it is now
+      .dispatching/
+         .factories/
+            . # the same as it is now
+      .formats/
+         .base/
+            .query_compiler.py # BaseQueryCompiler
+         .pandas/
+            .base/ # or default (DefaultPandasQueryCompiler)
+               .query_compiler.py # BasePandasQueryCompiler
+            .kernels.py
+            .query_compiler.py # PandasQueryCompiler
+            .parsers.py
+            .utils.py # data_management/utils moved here
+      .execution/
+        .ray/
+           .frame/
+              .dataframe.py # GenericRayDataframe
+              .metadata.py # GenericRayDataframeMetadata
+           .partition_management/ # or `partitioning`
+              .axis_partition.py # GenericRayDataframeAxisPartition
+              .partition_manager.py # GenericRayDataframeManager
+              .partition.py # GenericRayDataframePartition
+           .io/
+              .io.py # GenericRayIO
+           .task_wrapper.py # GenericRayTask
+           .utils.py # init_ray
+        .dask/
+           . # the same as for ray
+        .python/
+           . # the same as for ray
+      .dataframe/  # implementations go here
+         .algebra/
```

Comment:
Shouldn't we put base classes of Dataframe (BaseDataframe, etc.) here? For example, in `base` folder.

Reply:
`algebra` here is meant to be the general definitions of operators supported by Modin. The base classes have more implementation details, so I placed them there.

Reply:
Do you mean creating the following structure in `algebra` folder with no base (abstract) dataframe and metadata_management classes definitions?

```python
./default_methods # should probably be renamed to d2p (default to pandas). What do you think?
.map.py
.reduce.py
.tree_reduce.py
.explode.py
.groupby.py
.etc.
```